tests/sklearn/test_sklearn_models.py

```diff
@@ -198,33 +186,79 @@ def check_correctness_with_sklearn(
         "Ridge": 0.9,
         "ElasticNet": 0.9,
         "XGBRegressor": -0.2,
-        "NeuralNetRegressor": -10,
     }
 
-    # For classifiers
-    threshold_accuracy_dic = {
+    # For accuracy measures
+    threshold_accuracies = {
         "LogisticRegression": 0.9,
         "LinearSVC": 0.9,
         "XGBClassifier": 0.7,
         "RandomForestClassifier": 0.8,
-        "NeuralNetClassifier": 0.7,
         "KNeighborsClassifier": 0.9,
     }
 
     model_name = get_model_name(model_class)
-    acceptance_r2score = acceptance_r2score_dic.get(model_name, 0.9)
-    threshold_accuracy = threshold_accuracy_dic.get(model_name, 0.9)
+    acceptance_r2score = acceptance_r2scores.get(model_name, 0.9)
+    threshold_accuracy = threshold_accuracies.get(model_name, 0.9)
+
+    # If the model is a classifier
+    # KNeighborsClassifier does not provide a predict_proba method for now
+    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3962
+    if (
```

Comment:
the following tests all inference methods (predict, predict_proba, decision_function) and checks that results are close to sklearn